datalab/stackdriver/monitoring/_resource.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied.  See the License for the specific language governing permissions and limitations under
+# the License.
+
+"""Provides the ResourceDescriptors in the monitoring API."""
+
+from __future__ import absolute_import
+from builtins import object
+
+import collections
```

Comment:
There are lots of warnings when running `flake8`. In terms of the warnings, ` indentation is not a multiple of four` and  `line too long (x > 79 characters)` may not be enforced here, but perhaps the other warnings are useful.

```
tony@tonypc:~/pydatalab-parthea$ git diff master | flake8 --diff
./datalab/stackdriver/commands/_monitoring.py:17:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:19:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:65:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:69:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:70:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:71:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:72:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:76:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:77:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:78:3: E111 indentation is not a multiple of four
./datalab/stackdriver/commands/_monitoring.py:79:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/__init__.py:17:1: F401 'Aligner' imported but unused
./datalab/stackdriver/monitoring/__init__.py:17:1: F401 'Reducer' imported but unused
./datalab/stackdriver/monitoring/__init__.py:18:1: F401 'MetricDescriptors' imported but unused
./datalab/stackdriver/monitoring/__init__.py:19:1: F401 'ResourceDescriptors' imported but unused
./datalab/stackdriver/monitoring/__init__.py:20:1: F401 'Query' imported but unused
./datalab/stackdriver/monitoring/_metric.py:3:80: E501 line too long (98 > 79 characters)
./datalab/stackdriver/monitoring/_metric.py:8:80: E501 line too long (99 > 79 characters)
./datalab/stackdriver/monitoring/_metric.py:9:80: E501 line too long (98 > 79 characters)
./datalab/stackdriver/monitoring/_metric.py:10:80: E501 line too long (100 > 79 characters)
./datalab/stackdriver/monitoring/_metric.py:26:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:28:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:31:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:38:80: E501 line too long (82 > 79 characters)
./datalab/stackdriver/monitoring/_metric.py:49:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:60:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:65:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:71:80: E501 line too long (81 > 79 characters)
./datalab/stackdriver/monitoring/_metric.py:78:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:80:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:81:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_metric.py:87:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:3:80: E501 line too long (98 > 79 characters)
./datalab/stackdriver/monitoring/_resource.py:8:80: E501 line too long (99 > 79 characters)
./datalab/stackdriver/monitoring/_resource.py:9:80: E501 line too long (98 > 79 characters)
./datalab/stackdriver/monitoring/_resource.py:10:80: E501 line too long (100 > 79 characters)
./datalab/stackdriver/monitoring/_resource.py:18:1: F401 'collections' imported but unused
./datalab/stackdriver/monitoring/_resource.py:25:1: E302 expected 2 blank lines, found 1
./datalab/stackdriver/monitoring/_resource.py:26:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:28:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:30:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:44:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:55:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:60:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:66:80: E501 line too long (81 > 79 characters)
./datalab/stackdriver/monitoring/_resource.py:73:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:75:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:76:7: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_resource.py:80:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_visualization.py:3:80: E501 line too long (98 > 79 characters)
./datalab/stackdriver/monitoring/_visualization.py:8:80: E501 line too long (99 > 79 characters)
./datalab/stackdriver/monitoring/_visualization.py:9:80: E501 line too long (98 > 79 characters)
./datalab/stackdriver/monitoring/_visualization.py:10:80: E501 line too long (100 > 79 characters)
./datalab/stackdriver/monitoring/_visualization.py:19:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_visualization.py:28:3: E111 indentation is not a multiple of four
./datalab/stackdriver/monitoring/_visualization.py:30:3: E111 indentation is not a multiple of four
./tests/main.py:40:1: E402 module level import not at top of file
./tests/main.py:41:1: E402 module level import not at top of file
./tests/main.py:42:1: E402 module level import not at top of file
./tests/main.py:43:1: E402 module level import not at top of file
./tests/main.py:44:1: E402 module level import not at top of file
./tests/main.py:45:1: E402 module level import not at top of file
./tests/main.py:46:1: E402 module level import not at top of file
./tests/main.py:47:1: E402 module level import not at top of file
./tests/stackdriver/__init__.py:3:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/__init__.py:8:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/__init__.py:9:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/__init__.py:10:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/monitoring/__init__.py:3:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/monitoring/__init__.py:8:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/monitoring/__init__.py:9:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/monitoring/__init__.py:10:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/monitoring/metric_tests.py:3:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/monitoring/metric_tests.py:8:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/monitoring/metric_tests.py:9:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/monitoring/metric_tests.py:10:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/monitoring/metric_tests.py:31:80: E501 line too long (84 > 79 characters)
./tests/stackdriver/monitoring/metric_tests.py:32:80: E501 line too long (82 > 79 characters)
./tests/stackdriver/monitoring/metric_tests.py:39:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:43:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:44:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:58:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:74:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:75:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:87:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:88:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:103:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:104:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:115:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:116:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:127:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:128:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:150:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:151:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:152:5: F841 local variable 'NUM_METRICS' is assigned to but never used
./tests/stackdriver/monitoring/metric_tests.py:164:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:165:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:166:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:170:5: F841 local variable 'actual_table' is assigned to but never used
./tests/stackdriver/monitoring/metric_tests.py:175:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:176:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:180:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/metric_tests.py:181:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:3:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:8:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:9:80: E501 line too long (98 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:10:80: E501 line too long (99 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:28:80: E501 line too long (85 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:29:80: E501 line too long (80 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:32:1: E302 expected 2 blank lines, found 1
./tests/stackdriver/monitoring/resource_tests.py:34:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:38:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:39:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:52:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:66:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:67:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:77:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:78:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:79:7: E125 continuation line with same indent as next logical line
./tests/stackdriver/monitoring/resource_tests.py:92:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:93:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:94:7: E125 continuation line with same indent as next logical line
./tests/stackdriver/monitoring/resource_tests.py:103:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:104:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:105:80: E501 line too long (81 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:113:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:114:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:115:80: E501 line too long (82 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:132:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:133:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:134:80: E501 line too long (82 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:143:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:144:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:144:80: E501 line too long (80 > 79 characters)
./tests/stackdriver/monitoring/resource_tests.py:145:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:149:5: F841 local variable 'actual_table' is assigned to but never used
./tests/stackdriver/monitoring/resource_tests.py:154:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:155:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:159:3: E111 indentation is not a multiple of four
./tests/stackdriver/monitoring/resource_tests.py:160:3: E111 indentation is not a multiple of four
```